core/src/org/pentaho/platform/engine/services/connection/datasource/dbcp/NonPooledDatasourceSystemListener.java

```diff
@@ -66,6 +66,8 @@ public boolean startup( final IPentahoSession session ) {
           if ( !databaseConnection.getAccessType().equals( DatabaseAccessType.JNDI ) ) {
             // if connection's port used by server there is no sense to get DataSource for this
             ds = isPortUsedByServer( databaseConnection ) ? null : setupDataSourceForConnection( databaseConnection );
+          } else {
+            ds = setupDataSourceForConnection( databaseConnection );
```

Comment:
setupDataSourceForConnection doesn't seem to handle JNDI connections. Would it be better to simply not populate the cache with this datasource period?